packages/cardpay-sdk/sdk/token-bridge-foreign-side.ts

```diff
@@ -140,27 +171,38 @@ export default class TokenBridgeForeignSide implements ITokenBridgeForeignSide {
     return estimatedGasInWei.divRound(rounder).mul(rounder);
   }
 
+  async claimBridgedTokens(txnHash: string): Promise<TransactionReceipt>;
   async claimBridgedTokens(
     messageId: string,
     encodedData: string,
     signatures: string[],
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async claimBridgedTokens(
+    messageIdOrTxnHash: string,
+    encodedData?: string,
+    signatures?: string[],
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    if (!encodedData) {
+      return waitUntilTransactionMined(this.layer1Web3, messageIdOrTxnHash);
+    }
     let from = contractOptions?.from ?? (await this.layer1Web3.eth.getAccounts())[0];
     let foreignAmbAddress = await getAddress('foreignAMB', this.layer1Web3);
     let foreignAmb = new this.layer1Web3.eth.Contract(ForeignAMBABI as AbiItem[], foreignAmbAddress);
     let events = await foreignAmb.getPastEvents('RelayedMessage', {
       fromBlock: 0,
       toBlock: 'latest',
-      filter: { messageId: messageId },
+      filter: { messageId: messageIdOrTxnHash },
     });
     if (events.length === 1) {
       let txnHash = events[0].transactionHash;
       let receipt = await this.layer1Web3.eth.getTransactionReceipt(txnHash);
       return receipt;
     }
-    const packedSignatures = prepSignaturesForExecution(signatures);
+    const packedSignatures = prepSignaturesForExecution(signatures!);
```

Comment:
ditto with the comments above around using a test to assert this exists instead of using !'s